syntax = "proto3";
package Godot;
import "frame.proto";
import "entity.proto";
import "text_message.proto";

enum MessageType {
    TEXT = 0;
    FRAME = 1;
    ACTION = 2;
}

enum ActionType {
    CREATE_ENTITY = 0;
    DESTROY_ENTITY = 1;
}

message Action {
    ActionType type = 1;
    Entity entity = 2;
}

message GNSMessage {
    MessageType type = 1;
    TextMessage text = 2;
    Frame frame = 3;
    Action action = 4;
}

//import "google/protobuf/any.proto";
//message AnyMessage {
//    google.protobuf.Any content = 1;
//}
